app/services/MatcherPool.scala

```diff
@@ -75,24 +77,44 @@ class MatcherPool(val maxCurrentJobs: Int = 8, val maxQueuedJobs: Int = 1000, va
     Logger.info(s"Validation job with id: ${query.requestId} received. Checking categories: ${categoryIds.mkString(", ")}")
 
     val eventuallyResponses =
-      createJobsFromPartialJobs(checkStrategy(query.blocks, categoryIds), query.requestId)
+      createJobsFromPartialJobs(defaultCheckStrategy(query.blocks, categoryIds), query.requestId)
       .map(offerJobToQueue)
 
-    Future.sequence(eventuallyResponses).map { matches =>
+    Future.sequence(eventuallyResponses).map { results =>
       Logger.info(s"Validation job with id: ${query.requestId} complete")
-      matches.flatten
+      MatcherResponse(
+        results.flatMap(_.job.blocks).distinct,
+        results.flatMap(_.job.categoryIds).distinct,
+        results.flatMap(_.matches)
+      )
     }
   }
 
-  /**
-    * @see MatcherPoolEventBus
-    */
-  def subscribe(subscriber: MatcherPoolSubscriber): Boolean = eventBus.subscribe(subscriber, subscriber.requestId)
+  def check(query: Check): Future[MatcherResponse] = check(query, defaultCheckStrategy)
 
-  /**
-    * @see MatcherPoolEventBus
-    */
-  def unsubscribe(subscriber: MatcherPoolSubscriber): Boolean = eventBus.unsubscribe(subscriber, subscriber.requestId)
+  def checkStream(query: Check, checkStrategy: CheckStrategy): Source[MatcherResponse, NotUsed] = {
+    val categoryIds = query.categoryIds match {
+      case None => getCurrentCategories.map { case (_, category) => category.id }
+      case Some(ids) => ids
+    }
+
+    Logger.info(s"Validation job with id: ${query.requestId} received. Checking categories: ${categoryIds.mkString(", ")}")
+
+    val jobs = createJobsFromPartialJobs(checkStrategy(query.blocks, categoryIds), query.requestId)
+    Logger.info(s"Created ${jobs.size} jobs")
+    val eventualResponses = jobs map offerJobToQueue
+    val responsesAsSources = eventualResponses map Source.fromFuture
+    val mergedResponses = responsesAsSources.foldLeft(Source.empty[MatcherJobResult]) { case (source, acc) =>
+        acc.merge(source)
```

Comment:
Just FYI, I took a look and we can also create streams directly from lists of `Futures`, which makes this a one liner --

```
val responseStream = Source(eventualResponses).mapAsyncUnordered(1)(identity)
```

-- I can't see anything funny in the `mapAsyncUnordered` call re: performance or parallelism (there's no blocking work to do) but you may spot a reason this is unsuitable!